core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1052,7 +1053,9 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
       debug(s"Live brokers: ${controllerContext.liveBrokerIds.mkString(",")}")
 
       val partitionsToActOn = controllerContext.partitionsOnBroker(id).filter { partition =>
-        controllerContext.partitionReplicaAssignment(partition).size > 1 && controllerContext.partitionLeadershipInfo.contains(partition)
+        controllerContext.partitionReplicaAssignment(partition).size > 1 &&
+          controllerContext.partitionLeadershipInfo.contains(partition) &&
+          !topicDeletionManager.isTopicQueuedUpForDeletion(partition.topic)
```

Comment:
I think we also need to exclude the partitions being deleted in the replicatedPartitionsBrokerLeads(). Otherwise, the shutting down broker will keep retrying the controlled shutdown request on those partitions since their leaders will never be moved.

Reply:
For the partitions being deleted, the two behaviors are
1. If we exclude them in the response, then controlled shutdown will be allowed immediately even though the controller might need to issue StopReplica requests for deletion on the given broker, hence possibly blocking the topic deletion until the broker is brought up again.
2. If we instead include the partitions being deleted, the controller will block the controlled shutdown for a while (bound by controlled shutdown retries) and wait for topic deletion to finish. Once the topic deletion finishes, the controlled shutdown response will be empty. Or if the controlled shutdown retries are exhausted before the topic deletion can finish, the broker will be shutdown regardless of the response.
Hence I think the current behavior does not violate correctness. It's a matter of trade-off. Thoughts?

Reply:
Good analysis. This is my thought. Normally, topic deletion will happen very quickly. So, if we get to a state where the deletion of a topic is started, but not completed when a broker is being shut down, chances are that another broker is down, which prevents the topic deletion from completing. In this case, having the broker retry the controlled shutdown request won't be useful and only adds latency during shutdown.

Reply:
That makes sense. I've made the change. Thanks!